src/js/language_label.js

```diff
@@ -1,15 +1,176 @@
+  { id: "hr", name: "Croatian" },
+  { id: "hu", name: "Hungarian" },
+  { id: "hy", name: "Armenian" },
+  { id: "id", name: "Indonesian" },
+  { id: "is", name: "Icelandic" },
+  { id: "it", name: "Italian" },
+  { id: "ja", name: "Japanese" },
+  { id: "ja_kana", name: "Japanese Kana" },
+  { id: "ja-Latn", name: "Japanese Romanized" },
+  { id: "ja-Hira", name: "Japanese Hiragana" },
+  { id: "ka", name: "Georgian" },
+  { id: "kk", name: "Kazakh" },
+  { id: "kn", name: "Kannada" },
+  { id: "ko", name: "Korean" },
+  { id: "ko-Latn", name: "Korean Romanized" },
+  { id: "ku", name: "Kurdish" },
+  { id: "la", name: "Latin" },
+  { id: "lb", name: "Luxembourgish" },
+  { id: "lt", name: "Lithuanian" },
+  { id: "lv", name: "Latvian" },
+  { id: "mk", name: "Macedonian" },
+  { id: "mt", name: "Maltese" },
+  { id: "ml", name: "Malayalam" },
+  { id: "nl", name: "Dutch" },
+  { id: "no", name: "Norwegian" },
+  { id: "oc", name: "Occitan" },
+  { id: "pl", name: "Polish" },
+  { id: "pt", name: "Portuguese" },
+  { id: "rm", name: "Romansh" },
+  { id: "ro", name: "Romanian" },
+  { id: "ru", name: "Russian" },
+  { id: "sk", name: "Slovak" },
+  { id: "sl", name: "Slovenian" },
+  { id: "sq", name: "Albanian" },
+  { id: "sr", name: "Serbian" },
+  { id: "sr-Latn", name: "Serbian Romanized" },
+  { id: "sv", name: "Swedish" },
+  { id: "ta", name: "Tamil" },
+  { id: "te", name: "Telugu" },
+  { id: "th", name: "Thai" },
+  { id: "tr", name: "Turkish" },
+  { id: "uk", name: "Ukrainian" },
+  { id: "zh", name: "Chinese" },
+];
+
+function labelControlElement(tag, id) {
+  var element = document.createElement(tag);
+  element.id = id;
+  Object.assign(element.style, {
+    margin: "0 2.5px",
+    color: "#444",
+  });
+  return element;
+}
+
+var tf = null;
+
+langChanger.onclick = function () {
+  hide(langChanger);
+  show(langHeader);
+
+  if (tf == null) {
+    tf = new Tokenfield({
+      el: document.querySelector("#language-picker"), // Attach Tokenfield to the input element with class "text-input"
+      items: langCodes,
+      newItems: false,
+    });
+    document.querySelectorAll(".tokenfield").forEach((e) => {
+      Object.assign(e.style, {
+        height: "5em",
+        width: "20em",
+        "margin-bottom": "4em",
+        "margin-top": "0.3em",
+      });
+      e.setAttribute("placeholder", "hi");
+    });
+    tf.on("change", function () {
+      let items = tf.getItems();
+      let langCodes = [];
+      items.forEach((element) => langCodes.push(element.id));
+      let langQuery = langCodes.join(",");
+      let rawHash = window.location.hash.split("&")[0];
+      let langQueryString =
+        langCodes.length > 0 ? `&language=${langQuery}` : "";
+      var currentURL = `${window.location.protocol}//${window.location.host}${window.location.pathname}${rawHash}${langQueryString}`;
+      window.history.pushState({ path: currentURL }, "", currentURL);
+      updateLanguageLabel();
+      hotReloadMap();
```

Comment:
Similar to `upgradeLegacyHash()`, we should just change the URL hash and the rest of the application will update automatically, adding to the browser history:

```suggestion
      let hash = window.location.hash.substr(1); // omit #
      let searchParams = new URLSearchParams(hash);
      url.searchParams.set("language", langCodes.join(","));
      window.location.hash = `#${searchParams}`;
```